### ISSUE TITLE
no icon on taskbar onwayland

### ISSUE THREAD

**Issue by userofryzen** [kotatogram/kotatogram-desktop#389]
### Steps to reproduce

1. open kotatogram on wayland session( at least, I have not tested on xorg yet)

### Expected behaviour

having the icon from kotatogram in all places including taskbar

### Actual behaviour

1 open kotatogram
2. the icon of kotatogram is missing 
![imagen](https://user-images.githubusercontent.com/33232977/188254434-9d1a455f-9e20-412d-9597-e297940ce9d4.png)
But in tray there is the correct icon, I have test too the telegram icon and it's correct too.
In addition I have been a few days using and ever have worked such icon but.. Idk maybe it's not a bug or something. 

### Operating system

Manjaro stable on latest updates with plasma wayland

### Version of Kotatogram Desktop

1.4.9 from https://aur.archlinux.org/packages/kotatogram-desktop-bin

### Installation source

Other (unofficial) source

**Comment by ilya-fedin** [comment]
Can you try the build from GitHub?

**Comment by userofryzen** [comment]
qt.qpa.wayland: EGL not available
error: : cannot open
error: : cannot open
error: : cannot open
QOpenGLWidget: Failed to create context
qt.svg: Error while inflating gzip file: SVG format check failed

the logs of konsole

**Comment by userofryzen** [comment]
> Can you try the build from GitHub?

what build? I mean, what from those flat.. give me a link  sorry

**Comment by ilya-fedin** [comment]
https://github.com/kotatogram/kotatogram-desktop/releases

**Comment by userofryzen** [comment]
> https://github.com/kotatogram/kotatogram-desktop/releases

https://github.com/kotatogram/kotatogram-desktop/releases/download/k1.4.9/1.4.9-linux.tar.xz if you mean this exactly it gives me the same problem no icon on taskbar. The simple and "standard" icon for an app that has no icon on wayland that I see a few times more.

**Comment by userofryzen** [comment]
QT_QPA_PLATFORM=xcb kotatogram-desktop 
error: : cannot open
error: : cannot open
error: : cannot open
qt.svg: Error while inflating gzip file: SVG format check failed
QCoreApplication::postEvent: Unexpected null receiver
QObject::~QObject: Timers cannot be stopped from another thread
pure virtual method called
terminate called without an active exception
Abortado (`core' generado)
 Okey, I have tested with xcb(xwayland) and the icon is working so the problem resides only on the wayland side. It's a bug, are you using wayland session? or maybe xorg one?

**Comment by ilya-fedin** [comment]
> if you mean this exactly it gives me the same problem no icon on taskbar.

But you say you're using kotatogram-desktop-bin, not the binary from tar.xz

**Comment by userofryzen** [comment]
> > if you mean this exactly it gives me the same problem no icon on taskbar.
> 
> But you say you're using kotatogram-desktop-bin, not the binary from tar.xz

the aur package is one that I have tested before and gives me the same problem, I have tested the another ones that is build on github like you said, from the URL I gave to you and.. the Same problem. 
can't you reproduce the problem?
The problem happens on wayland plasma session but not on xorg.

**Comment by ilya-fedin** [comment]
> the aur package is one that I have tested before and gives me the same problem, I have tested the another ones that is build on github like you said, from the URL I gave to you and.. the Same problem.

I don't quite understand, have you checked just unarchiving and launching AppImage or not? Without any AUR packages.

> can't you reproduce the problem?

It's hard for me to setup environment to reproduce. Would be nice if you tested the archive. Any re-packages like kotatogram-desktop-bin aren't the same.

**Comment by userofryzen** [comment]
> > the aur package is one that I have tested before and gives me the same problem, I have tested the another ones that is build on github like you said, from the URL I gave to you and.. the Same problem.
> 
> I don't quite understand, have you checked just unarchiving and launching AppImage or not? Without any AUR packages.
> 
> > can't you reproduce the problem?
> 
> It's hard for me to setup environment to reproduce. Would be nice if you tested the archive. Any re-packages like kotatogram-desktop-bin aren't the same.

yeah I have done that without any aur package.

I have tested both of them, the repackage from aur and the another one from github, both of them make me the same thing, no icon on taskbar

**Comment by ilya-fedin** [comment]
Can you provide the log.txt from application data directory?

**Comment by ilya-fedin** [comment]
Oh, forgot to mention that you need clean the data before launching the AppImage for it to detect first launch and integrate in the system. Otherwise such problems are expected.

**Comment by userofryzen** [comment]
@ilya-fedin how to clean the data ? so I can test it for you but the expected thing for me is that such problem is yet here. 
But tell me and Ill test. 
But pls tell me everything so I can do all the things at the same time

**Comment by ilya-fedin** [comment]
you can open the folder by entering `viewlogs` [cheat code](https://github.com/telegramdesktop/tdesktop/wiki/Cheat-Codes)

**Comment by userofryzen** [comment]
yeah, cleaning was the fix for the github one. Idk why on wayland fails such package. Ill report to aur then. 
I ll be using another one that is https://aur.archlinux.org/packages/kotatogram-desktop-beta-dynamic-bin
personally I don't see differents. Wich I should use ?

**Comment by ilya-fedin** [comment]
On Wayland, there's no such thing as 'window icon', so compositors just display the icon from the desktop file. If the package doesn't have the icon, it means desktop file name is not synchronized with what application expects. What application expects is logged to the log.txt.

**Comment by userofryzen** [comment]
I have reported it so.. let's wait. 
Thanks!

**Comment by userofryzen** [comment]
[log.txt](https://github.com/kotatogram/kotatogram-desktop/files/9485411/log.txt)
Flatpak stable package the same error as shown before, no icon.

**Comment by userofryzen** [comment]
[log.txt](https://github.com/kotatogram/kotatogram-desktop/files/9485415/log.txt)
The aur ones that is called kotatogram-desktop-bin
No icon either

**Comment by ilya-fedin** [comment]
Are you sure flatpak is properly integrated into your system? I.e. do you see Kotatogram item in the system menu?

**Comment by ilya-fedin** [comment]
> The aur ones that is called kotatogram-desktop-bin

According to the log, its .desktop file should have name `appimagekit_d2192f48ebc43a9db26e1dfa2bc5097b-Kotatogram_Desktop.desktop` in order for the integration to work.

**Comment by userofryzen** [comment]
[log.txt](https://github.com/kotatogram/kotatogram-desktop/files/9485418/log.txt)
the ones that is downloaded by the official website. 
And Ill see again the flatpak wait a moment

**Comment by userofryzen** [comment]
> Are you sure flatpak is properly integrated into your system? I.e. do you see Kotatogram item in the system menu?

No, there is not because it's appearing a general window icon on both tray and taskbar

**Comment by userofryzen** [comment]
> > The aur ones that is called kotatogram-desktop-bin
> 
> According to the log, its .desktop file should have name `appimagekit_d2192f48ebc43a9db26e1dfa2bc5097b-Kotatogram_Desktop.desktop` in order for the integration to work.

Okey.. then.. if I rename .desktop to that name it should work?!
With every update I should do that, right?

**Comment by ilya-fedin** [comment]
> No, there is not

That's the cause

**Comment by userofryzen** [comment]
> > No, there is not
> 
> That's the cause

how can I integrate?

I see that there is oppengl logs but disabled on the beta program, why?

**Comment by ilya-fedin** [comment]
> if I rename .desktop to that name it should work?!

Yeah, but it's weird it chooses such a name, like /etc/kotatogram-desktop/externalupdater doesn't exist or doesn't contain path to the binary, but I see the package install it

**Comment by userofryzen** [comment]
[2022.09.04 22:01:28] OpenGL: [FALSE] (OverlayWidget) this

**Comment by userofryzen** [comment]
> > if I rename .desktop to that name it should work?!
> 
> Yeah, but it's weird it chooses such a name, like /etc/kotatogram-desktop/externalupdater doesn't exist or doesn't contain path to the binary, but I see the package install it

okey, how should you fix on aur then?
I'll say in comments

**Comment by ilya-fedin** [comment]
> okey, how should you fix on aur then?

I think there's a bug in tdesktop code...

**Comment by userofryzen** [comment]
> tdesktop

May be.. Idk how it works to be honest. But I think I have discovered a bug here.
[2022.09.04 22:05:04] OpenGL: Could not create widget in a window.

While on 1.4.8 there is such opengl support I think. Intel here

**Comment by ilya-fedin** [comment]
> how can I integrate?

It's usually done automatically, but may be broken if you use not bash as your login shell as those integration scripts reside in /etc/profile.d and not all shells read /etc/profile

**Comment by userofryzen** [comment]
I have got lost with that message to be honest ilya.

**Comment by ilya-fedin** [comment]
I can't help with opengl, sorry, we haven't touched opengl code in the fork

**Comment by userofryzen** [comment]
okey, so bug upstream I think. 
But.. with tray icon then the best way to make it works it's to rename the desktop file , right?

But when you refer the bug on updater.. what you mean?

**Comment by ilya-fedin** [comment]
I see the logic that sets desktop file name doesn't work right due to wrong variable initialization order in upstream code

**Comment by userofryzen** [comment]
Okey, perfect, I hope the mantainer can look at it. 

Something weird that I have experienced now is that the https://aur.archlinux.org/packages/kotatogram-desktop-beta-dynamic-bin package initializes opengl .. I mean.. create the opengl context. It's something very weird , right? Because this package is the same as beta ones from you github...

**Comment by ilya-fedin** [comment]
No, it's not weird, this may be a bug in Qt version used in the AppImage

**Comment by userofryzen** [comment]
[log.txt](https://github.com/kotatogram/kotatogram-desktop/files/9485443/log.txt)
From your github 
[log.txt](https://github.com/kotatogram/kotatogram-desktop/files/9485444/log.txt)
directly. 
From aur package

amm Okay! I though that the appImage file for aur and github would be the same

**Comment by ilya-fedin** [comment]
> Okey, perfect, I hope the mantainer can look at it.

I don't think maintainer can do anything, it's a bug in the code, I've sent a PR upstream, but it will be a long time for the Kotatogram update

**Comment by ilya-fedin** [comment]
> appImage file for aur

kotatogram-desktop-beta-dynamic-bin doesn't use appimage I believe

**Comment by userofryzen** [comment]
thanks for all, let's wee whats happen
